MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoPresenter.swift

```diff
@@ -4,23 +4,54 @@ final class HomePageTwoPresenter {
     private let useCase: MoviesUseCaseProtocol!
     private let router: AppRouter!
     
+    var categories: [CategoryViewModel]!
+    
     init (useCase: MoviesUseCaseProtocol, router: AppRouter) {
         self.useCase = useCase
         self.router = router
+        
+        categories = []
     }
     
     func setDelegate(delegate: HomePageTwoDelegate) {
         self.delegate = delegate
     }
     
+    func getAllCategories() {
+        getPopularMovies()
+        getTrendingMovies()
+        getTopRatedMovies()
+    }
+    
+    func getMovies(for subcategory: LocalSubcategory) -> [MovieViewModel] {
+        var movies: [MovieViewModel] = []
+        categories.forEach {
+            if $0.subcategoryMovies.keys.contains(subcategory) {
+                movies = $0.subcategoryMovies[subcategory] ?? []
+            }
+        }
+        return movies
+    }
+    
+    func getSubcategories(for category: LocalCategory) -> [LocalSubcategory] {
+        var subcategories: [LocalSubcategory] = []
+        categories.forEach {
+            if $0.categoryKey == category {
+                subcategories = Array($0.subcategoryMovies.keys)
+            }
+        }
```

Comment:
If there's only one category in `categories` that will match this condition you should break the loop.
Also, you can use `first(where: )`